ref.go

```diff
@@ -111,15 +119,18 @@ func (e *Encoder) checkRefMap(v reflect.Value) (int, bool) {
 	}
 
 	if elem, ok := e.refMap[addr]; ok {
-		// the array addr is equal to the first elem, which must ignore
 		if elem.kind == kind {
-			return elem.index, ok
+			if elem.kind != reflect.Struct {
```

Comment:
is it a map or list if not struct?

Reply:
Yes, and also may include `reflect.Invalid`.